crates/agglayer-jsonrpc-api/src/service/mod.rs

```diff
@@ -0,0 +1,170 @@
+use agglayer_telemetry::KeyValue;
+use ethers::{providers::Middleware, types::H256};
+use futures::future::try_join;
+use tracing::{debug, error, info, instrument};
+
+pub use self::error::{CertificateRetrievalError, SendTxError, TxStatusError};
+use crate::{kernel::Kernel, signed_tx::SignedTx};
+
+pub mod error;
+
+/// The RPC agglayer service implementation.
+pub struct AgglayerService<Rpc> {
+    kernel: Kernel<Rpc>,
+}
+
+impl<Rpc> AgglayerService<Rpc> {
+    /// Create an instance of the RPC agglayer service.
+    pub fn new(kernel: Kernel<Rpc>) -> Self {
+        Self { kernel }
+    }
+}
+
+impl<Rpc> Drop for AgglayerService<Rpc> {
+    fn drop(&mut self) {
+        info!("Shutting down the agglayer service");
```

Comment:
The log message here is identical to the one in the `agglayer_rpc` crate. It could be differentiated by the module name but still might be confusing. How about something like the following?
```suggestion
        info!("Shutting down the legacy agglayer service");
```